app/controllers/api/v1/messages_controller.rb

```diff
@@ -0,0 +1,24 @@
+module Api
+  module V1
+    class MessagesController < Api::V1::ApiController
+
+      def index
+        @messages = current_user.matches.find(match_id).messages.includes(:user)
+      end
+
+      def create
+        @message = current_user.messages.create!(message_params.merge(match_id: match_id))
+      end
+
+      private
+
+      def message_params
+        params.require(:message).permit(:content)
+      end
+
+      def match_id
+        params[:match_id]
```

Comment:
I guess you can do `current_user.matches.find(params[:match_id])` here and change in the index to 
`match.messages.includes(:user)` and in the `create` merge with the record instead of the id, also it will prevent you from create a message in a match where you are not involved